File: src/main/java/mods/betterfoliage/client/gui/ConfigGuiFactory.java
package mods.betterfoliage.client.gui;

import java.util.Set;

import mods.betterfoliage.BetterFoliage;
import mods.betterfoliage.common.config.Config;
import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.GuiScreen;
import net.minecraftforge.fml.client.IModGuiFactory;
import net.minecraftforge.fml.client.config.GuiConfig;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

import com.google.common.collect.ImmutableSet;

@SideOnly(Side.CLIENT)
public class ConfigGuiFactory implements IModGuiFactory {

	public void initialize(Minecraft minecraftInstance) {
		
	}

	public Class<? extends GuiScreen> mainConfigGuiClass() {
		return ConfigGuiBetterFoliage.class;
	}

	public Set<RuntimeOptionCategoryElement> runtimeGuiCategories() {
		return ImmutableSet.<RuntimeOptionCategoryElement>of();
	}

	public RuntimeOptionGuiHandler getHandlerFor(RuntimeOptionCategoryElement element) {
		return null;
	}

	public static class ConfigGuiBetterFoliage extends GuiConfig {
		public ConfigGuiBetterFoliage(GuiScreen parentScreen) {
			super(parentScreen, Config.getConfigRootElements(), BetterFoliage.MOD_ID, null, false, false, BetterFoliage.MOD_NAME);
		}
	}
}

File: src/main/kotlin/mods/betterfoliage/client/gui/ConfigGuiFactory.kt
package mods.betterfoliage.client.gui

import mods.betterfoliage.BetterFoliageMod
import mods.betterfoliage.client.config.Config
import net.minecraft.client.Minecraft
import net.minecraft.client.gui.GuiScreen
import net.minecraftforge.fml.client.IModGuiFactory
import net.minecraftforge.fml.client.config.GuiConfig

class ConfigGuiFactory : IModGuiFactory {

    override fun initialize(minecraftInstance: Minecraft?) { }
    override fun hasConfigGui() = true
    override fun runtimeGuiCategories() = hashSetOf<IModGuiFactory.RuntimeOptionCategoryElement>()
    override fun createConfigGui(parentScreen: GuiScreen?) = createBFConfigGui(parentScreen)

    companion object {
        @JvmStatic
        fun createBFConfigGui(parentScreen: GuiScreen?) = GuiConfig(
            parentScreen,
            Config.rootGuiElements,
            BetterFoliageMod.MOD_ID,
            null,
            false,
            false,
            BetterFoliageMod.MOD_NAME
        )
    }
}
